### ISSUE TITLE
Update "spentness" after RPC calls to ClaimMarketDeposit, ClaimFeeDeposit, WithdrawMarket & TradeComplete

### ISSUE THREAD

**Issue by tiero** [TDex-network/tdex-daemon#298]
With #296 and #297 in place we need to define moments that will update the Unspent repository with either newly found unspent and to mark them as spent﻿

**Comment by tiero** [comment]
Actually I forgot that there is another case where "spentness" could be changed: The trader is not forced to call the `TradeComplete`, he could by himself broadcast the transaction right away. 

We [should still use the crawler to go fetch a txid after TradePropose ](https://github.com/TDex-network/tdex-daemon/blob/master/internal/core/application/trade_service.go#L401) and therefore we could have the crawler to still update the unspents repository.

My idea is to not watch continuously, but just go watch one time only after the [ExpiryTime](https://github.com/TDex-network/tdex-daemon/blob/master/internal/core/application/trade_service.go#L380): If the tx is NOT in mempool by then, we can leave it to the TTL to "unlock" the unspents, otherwise we go to mark as spent that linked utxos.

**Comment by altafan** [comment]
> My idea is to not watch continuously, but just go watch one time only after the ExpiryTime: If the tx is NOT in mempool by then, we can leave it to the TTL to "unlock" the unspents, otherwise we go to mark as spent that linked utxos.

what is supposed to be this "one time after the ExpiryTime"? What if in this time frame the trade service receives a proposal request?
In this way, we might risk making the daemon to try double-spending utxos that are actually spent while they would be recognized as unspent simply because the bc listener hasn't done its job yet.

**Comment by tiero** [comment]
a) Trade `A` gets accepted at `t0`
b) it locks utxo `X` taken from the utxo set `U` with expiry time `t5`
c) a ticker is attached to trade `A` txid's to be fired a `t5+` to go check on chain if it exists in mempool
d) new utxo set becomes  `N` which is `U - X`
e) Trade `B` comes in and gets accepted at `t1`
f) it locks utxo `Y` from utxo set `N`  with expiry time `t6`
g) a ticker is attached to trade `B` txid's to be fired a `t6+` to go check on chain if it exists in mempool
h) new utxo set becomes  `M` which is `N - Y`
h) we are `t6` exiprty time for trade `A`, the ticker fire up and goes on chain and finds nothing: we can unlock the utxo `X` locked by trade `A` so we have utxo set `L` which is `M + X`


The daemon now can and should double spend the utxo X to be sure some sort of replay attack by trader of trade A (but this is another story)

**Comment by altafan** [comment]
> a) Trade A gets accepted at t0
b) it locks utxo X taken from the utxo set U with expiry time t5
c) a ticker is attached to trade A txid's to be fired a t5+ to go check on chain if it exists in mempool
d) new utxo set becomes N which is U - X

So, let's say `A` is accepted at t0, and that the trader broadcasts the completed trade by himself at `t5.5`.
From the trader pov this should likely be fine.. I'm not talking about long periods of time after the expiration (in those cases it would be ok if the inputs of `A` would have been used for other trades), but rather of acceptable delays, like within a minute after the expiration. 

> d) new utxo set becomes N which is U - X
e) Trade B comes in and gets accepted at t1
f) it locks utxo Y from utxo set N with expiry time t6


We are now at `t5.2`, when let's say comes `B` (instead of `t6`).. so a little bit before `A` gets settled but still after the unspent unlock. At this time, the utxo set would actually be `U` because the lock on `X` has been released `t5` by following the flow you defined. If exactly that utxo is selected for `B`, the trade will be recognized as trying to double spend a utxo when broadcasted. Until that time, the daemon will likely recognize it as valid.

**Comment by tiero** [comment]
> If exactly that utxo is selected for B, the trade will be recognized as trying to double spend a utxo when broadcasted
Let me rephrase 

m) trade `C` comes in at `t5.2` and locks the new available `X` (cause unlocked by  TTL)
n) ) a ticker is attached to trade `C` txid's to be fired a t10.2+ to go check on chain if it exists in mempool
o) new utxo set becomes `O` which is M - X
p) user correctly broadcast at `t8` let's say, before daemon coudl double-spend un expired trade
q) it does the job for us, goint into block before the trader that may try to settle after the expiry
